aptos-move/e2e-move-tests/src/tests/resource_groups.data/pack/sources/resource_groups_test.move

```diff
@@ -0,0 +1,237 @@
+module 0x1::resource_groups_test {
+    use aptos_std::table::{Self, Table};
+    use std::vector;
+    use std::string::{Self, String};
+    use aptos_framework::account;
+    use std::signer;
+
+    const USE_RESOURCE_TYPE: u32 = 0;
+    const USE_TABLE_TYPE: u32 = 1;
+    const USE_RESOURCE_GROUP_TYPE: u32 = 2;
+
+    /// When checking the value of aggregator fails.
+    const ENOT_EQUAL: u64 = 17;
+
+    const EINVALID_ARG: u64 = 18;
+
+    const ERESOURCE_DOESNT_EXIST: u64 = 19;
+    const ETABLE_DOESNT_EXIST: u64 = 20;
+    const ERESOURCE_GROUP_DOESNT_EXIST: u64 = 21;
+    const EINDEX_DOESNT_EXIST: u64 = 22;
+    const EOPTION_DOESNT_EXIST: u64 = 23;
+
+    #[resource_group(scope = global)]
+    struct MyGroup {}
+
+    // TODO: I choose all these resources to have different structure to have a more comprehensive testing.
+    // But this also means, I needed to have individual set functions for each resource. Is that okay?
+    // Or should I have same structure for all the resources?
```

Comment:
I think it is perfectly fine to have all resources have a string field (potentially at different indices), which is set via the function, and have the rest be different, and just be initialized.

we are generally testing behavior across resources, it is unlikely a single resource itself might have issues.


and then you don't need different set methods.


you also probably need a check method - i.e. comparison to a value, and aborting if not matching, and then potentially some methods that touch two things like set_and_check(index1, index2)

Reply:
Done.